tests/test_aggregate.py

```diff
@@ -91,34 +93,106 @@ class Repos(NamedTuple):
+                self.id,
+                (self.id == 666),
+            )
+            return self._has_failures
+
+        def __repr__(self):
+            return f"<MockIncident id: {self.id} with repos {self.channels}>"
+
+    def _func(product, version, arch, embargoed=False, mocked_incident=42):
         repo = Repos(product=product, version=version, arch=arch)
-        return MockIncident(repo, embargoed=embargoed)
+        return MockIncident(repo, embargoed=embargoed, mocked_incident=mocked_incident)
 
     return _func
 
 
-def test_aggregate_call_with_test_issues(request_mock, incident_mock, monkeypatch):
+from logging import getLogger
+
+logger = getLogger(__name__)
+
+
+def mock_has_failures(self, token):
+    logger.info(
+        "has failures called for incident %s, evaluated to %s",
+        self.id,
+        (self.id == 666),
+    )
+    return self.id == 666
+
+
+def test_aggregate_call_with_test_issues(
+    request_mock, incident_mock, monkeypatch, caplog
+):
     """
     Test with a valid incident
     """
+    # monkeypatch.setattr(Incident, "has_failures", mock_has_failures)
+    # monkeypatch.setattr(Incident, "_rev", lambda self, *args: 42)
+    caplog.set_level("DEBUG")
+
     my_config = {}
-    my_config["FLAVOR"] = None
-    my_config["archs"] = ["ciao"]
+    my_config["FLAVOR"] = "Habanero"
+    my_config["archs"] = ["ciao", "aarch64"]
     my_config["test_issues"] = {"AAAAAAA": "BBBBBBBBB:CCCCCCCC"}
+    my_config["product"] = "openQA"
     acc = Aggregate("", settings={}, config=my_config)
+
+    these_incidents = []
+    good_incident = incident_mock(
+        product="BBBBBBBBB", version="CCCCCCCC", arch="ciao", mocked_incident=1284
+    )
+    these_incidents.append(good_incident)
+
     res = acc(
-        incidents=[incident_mock(product="BBBBBBBBB", version="CCCCCCCC", arch="ciao")],
+        these_incidents,
         token=None,
         ci_url=None,
     )
+
     assert len(res) == 1
+    logger.info("Testing the the_demon_incident (%s)", len(res))
+
+    logger.info("Testing the the_demon_incident (%s)", len(res))
+    [logger.info("res %s", r) for r in res[0]["qem"]["incidents"]]
+    the_demon_incident = incident_mock(
+        product="BBBBBBBBB", version="CCCCCCCC", arch="ciao", mocked_incident=666
+    )
+    these_incidents.append(the_demon_incident)
+
+    res = acc(
+        these_incidents,
+        token=None,
+        ci_url=None,
+    )
+
+    assert len(res) == 1  # the demon incident is not added, because it has failures
+
+    the_moved_incident = incident_mock(
+        product="BBBBBBBBB", version="CCCCCCCC", arch="ciao", mocked_incident=1335
+    )
+    these_incidents.append(the_moved_incident)
+    logger.info("Testing the the_moved_incident (%s)", len(res))
+
+    res = acc(
+        these_incidents,
+        token=None,
+        ci_url=None,
+    )
+
+    assert len(res) == 2  # the demon incident is not added, because it has failures
```

Comment:
I am not sure what you are asserting here. This is where https://github.com/openSUSE/qem-bot/actions/runs/7724416299/job/21056528243?pr=154#step:5:178 fails. Your comment says "the demon incident is not added, because it has failures" which is what https://github.com/openSUSE/qem-bot/pull/154/files#diff-09e8e60e6d3ec2a94d9f6571f313fef2ea2e181b40892485011bea95c7e11f66R181 already stated. But you assume that the "moved incident" is then included, right? Unfortunately I don't have further hints. I would use the debugger to step through the execution here as my experience with this code is also quite limited.

Reply:
> I am not sure what you are asserting here. This is where https://github.com/openSUSE/qem-bot/actions/runs/7724416299/job/21056528243?pr=154#step:5:178 fails. Your comment says "the demon incident is not added, because it has failures" which is what https://github.com/openSUSE/qem-bot/pull/154/files#diff-09e8e60e6d3ec2a94d9f6571f313fef2ea2e181b40892485011bea95c7e11f66R181 already stated. But you assume that the "moved incident" is then included, right? Unfortunately I don't have further hints. I would use the debugger to step through the execution here as my experience with this code is also quite limited.

Problem was that `res` is a list of whatever, but the information about the incidents was hidden in the elements of it, so `res[0][qem][incidents]` was the code I was looking for

so all this time, the incidents were like:

![grafik](https://github.com/openSUSE/qem-bot/assets/229240/5f4151c4-cc37-43fd-a4ec-960f68a2780e)